m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -118,23 +122,31 @@
 
     private final Set<MetricTag> commonTags;
 
-    // Holds metrics to be flushed
-    private final BlockingQueue<SizedMetric> metricQueue;
-    // The service used to execute metric flushes
-    private final ExecutorService executor;
+    // TODO replace w/ an evicting queue
+    private final BlockingQueue<SizedMetric> queue;
+
+    // Executor service running processors flushing metrics to collectors
+    private final ExecutorService executorService;
+
+    private final ScheduledExecutorService scheduledExecutorService;
```

Comment:
Why do we need both ExecutorService and ScheduledExecutorService? Can you add documentation on what should be the usage for either of them?

Reply:
First runs processors (1 thread per processor), the second -- periodic tasks. I can add documentation, but not sure how much that makes sense given that we will be revisiting the library altogether.